### PR TITLE
Book Rental Functionality

### PR DESCRIPTION
**Pull request by zeminlu**
## Summary
The book rental view, buttons and functionality.

## Screenshots
screen shot 2016-01-20 at 3 13 57 pm
screen shot 2016-01-20 at 3 15 22 pm
screen shot 2016-01-20 at 3 15 26 pm

### CODE REVIEW

File: src/app/components/site/private/books/bookController.js [hunk 1]
```diff
@@ -21,5 +22,14 @@ angular.module('wbooks').controller('BookController', [
         this.failedFetchingBookErrorCode = reason.data.code;
         this.failedFetchingBookErrorMessage = reason.data.error;
       });
+
+    this.goRentBook = (bookId) => {
```
Comment (sbalay) [inline src/app/components/site/private/books/bookController.js]: use url params in the state and solve this with ui-sref

File: src/app/components/site/private/books/bookRentalController.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
+angular.module('wbooks').controller('BookRentalController', [
+  'BooksService',
+  'localStorageService',
+  '$rootScope',
+  '$state',
+  function (BooksService, localStorageService, $rootScope, $state) {
+    this.backToList = () => {
```
Comment (sbalay) [inline src/app/components/site/private/books/bookRentalController.js]: where are we using this function?

File: src/app/components/site/private/books/bookRentalController.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      if (this.page) {
+        return true;
+      }
+      return false;
+    };
+
+    this.backToDetail = () => {
```
Comment (sbalay) [inline src/app/components/site/private/books/bookRentalController.js]: where are we using this function?

File: src/app/components/site/private/books/bookRentalController.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+      if (this.page) {
+        return false;
+      }
+      return true;
+    };
+
+    this.setBookId = () => {
```
Comment (sbalay) [inline src/app/components/site/private/books/bookRentalController.js]: state params

File: src/app/components/site/private/books/bookRentalController.js [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+          this.failedRentingBook = true;
+          this.failedRentingBookErrorCode = reason.data.code;
+          this.failedRentingBookErrorMessage = reason.data.error;
+        }
+      );
+    }
+
+    this.setBookId();
+
+    this.rentPeriod = 15;
```
Comment (sbalay) [inline src/app/components/site/private/books/bookRentalController.js]: avoid declaring variables and methods that are not used in the view in the controller scope. instead use ```const rentPeriod = 15```

File: src/app/components/site/private/books/bookRentalController.js [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+      opened: false
+    };
+
+    this.setEndDate = () => {
+      this.rentalData.to.iso = new Date(this.rentalData.from.iso.valueOf());
+      this.rentalData.to.iso.setDate(this.rentalData.to.iso.getDate() + this.rentPeriod);
+    };
+
+    if (localStorageService.isSupported) {
+      const page = localStorageService.get('booksPage');
```
Comment (sbalay) [inline src/app/components/site/private/books/bookRentalController.js]: use state params for the pagination data

File: src/app/components/site/private/books/booksDashboardController.js [hunk 7]
```diff
@@ -89,11 +89,23 @@ angular.module('wbooks').controller('BooksDashboardController', [
...
+        localStorageService.set('bookRentId', bookId);
+      } else if (localStorageService.cookie.isSupported) {
+        localStorageService.cookie.set('bookRentId', bookId);
+      }
+      if (localStorageService.isSupported) {
+        localStorageService.set('booksPage', this.page);
+      } else if (localStorageService.cookie.isSupported) {
+        localStorageService.cookie.set('booksPage', this.page)
+      }
+      $state.go('site.private.bookrental');
```
Comment (sbalay) [inline src/app/components/site/private/books/booksDashboardController.js]: state params

File: src/app/components/site/private/books/bookRentalController.js (hunk 3 above)
Comment (sbalay) [inline src/app/components/site/private/books/bookRentalController.js]: :point_up:

File: src/app/components/site/private/books/bookRentalController.js [hunk 8]
```diff
@@ -0,0 +1,86 @@
+angular.module('wbooks').controller('BookRentalController', [
+  'BooksService',
+  '$rootScope',
+  '$state',
+  '$stateParams',
+  function (BooksService, $rootScope, $state, $stateParams) {
+    this.backToList = () => {
```
Comment (sbalay) [inline src/app/components/site/private/books/bookRentalController.js]: Where are we using this?

Comment (zeminlu) [inline src/app/components/site/private/books/bookRentalController.js]: @sbalay  Both functions were left there by mistake, were intended for something I didn't use after all.

**Comment by sbalay** [comment]
:+1:

**Pull request merged**